multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/ValidateDeployParametersStep.java

```diff
@@ -181,21 +185,31 @@ private void mergeArchive(ProcessContext context, List<FileEntry> archivePartEnt
                                                        archivePartEntries.size(), archiveSize);
             FileEntry uploadedArchive = persistArchive(archiveStreamWithName, context, archiveSize);
             context.setVariable(Variables.APP_ARCHIVE_ID, uploadedArchive.getId());
+            validateMergedArchive(archiveStreamWithName);
             getStepLogger().infoWithoutProgressMessage(MessageFormat.format(Messages.ARCHIVE_WITH_ID_0_AND_NAME_1_WAS_STORED,
                                                                             uploadedArchive.getId(),
                                                                             archiveStreamWithName.getArchiveName()));
+        } catch (IOException e) {
+            throw new SLException(e);
         } finally {
             IOUtils.closeQuietly(archiveStreamWithName.getArchiveStream());
         }
     }
 
+    private void validateMergedArchive(ArchiveStreamWithName archiveStreamWithName) throws IOException {
+        LazyArchiveInputStream lazyArchiveInputStream = (LazyArchiveInputStream) archiveStreamWithName.getArchiveStream();
```

Comment:
This is the already read stream. Open a new stream via:
fileService::openInputStream
use try-catch or finally